contracts/staking/ReceiverStaking.sol

```diff
@@ -148,16 +148,45 @@ contract ReceiverStaking is
         currentEpoch = _getCurrentSnapshotId();
     }
 
+    function getCurrentEpoch() public view returns (uint256) {
+        return _getCurrentSnapshotId();
+    }
+
     function balanceOfSignerAt(address signer, uint256 snapshotId) public view returns (uint256 balance, address account) {
         account = signerToStaker[signer];
         balance = ERC20SnapshotUpgradeable.balanceOfAt(account, snapshotId);
     }
 
+    function balanceOfSignerAtRanged(address signer, uint256 _from, uint256 _count) public view returns (uint256[] memory balances, address account) {
+        account = signerToStaker[signer];
+        balances = new uint256[](_count);
+        uint256 i = 0;
+        while (i < _count) {
+            balances[i] = balanceOfAt(account, _from);
+            unchecked {
+                ++_from;
+                ++i;
```

Comment:
can probably skip this, if we change i to last and change condition to from < last

Reply:
same thing in `getClustersRanged` and `totalSupplyAtRanged`

Reply:
Ahh.. right. Skip then....